WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -313,7 +313,7 @@ def my_competitions
     if current_user.person
       @competitions += current_user.person.competitions
     end
-    @competitions = @competitions.uniq.sort_by(&:start_date).reverse!
+    @competitions = @competitions.uniq.reject(&:start_date) + @competitions.uniq.select(&:start_date).sort_by(&:start_date).reverse!
```

Comment:
Something like this should work:

`@competitions = @competitions.uniq.sort_by { |comp| comp.start_date || Date.new(2100, 1, 1) }.reverse`

I tried getting rid of the `reverse`, but it's weirdly hard to sort date in reverse order.